modular_skyrat/modules/liquids/code/reagent_containers.dm

```diff
@@ -0,0 +1,31 @@
+/obj/item/reagent_containers/Initialize(mapload, vol)
+	. = ..()
+
+	AddComponent(/datum/component/liquids_interaction, TYPE_PROC_REF(/obj/item/reagent_containers/cup/beaker, attack_on_liquids_turf))
+
+/obj/item/reagent_containers/proc/attack_on_liquids_turf(obj/item/reagent_containers/my_beaker, turf/T, mob/living/user, obj/effect/abstract/liquid_turf/liquids)
+	if(user.combat_mode)
+		return FALSE
+	if(!my_beaker.spillable)
+		return FALSE
+	if(!user.Adjacent(T))
+		return FALSE
+	if(liquids.fire_state) //Use an extinguisher first
+		to_chat(user, "<span class='warning'>You can't scoop up anything while it's on fire!</span>")
+		return TRUE
+	if(liquids.height == 1)
```

Comment:
(Not asking you to change this unless you want to) good god the inability to fully clean spills sucks

Reply:
The mechanic as designed is that deep spills are scooped with reagent containers and then puddles with a mop.